python/taichi/ui/staging_buffer.py

```diff
@@ -93,25 +94,81 @@ def copy_colors_to_vbo(vbo, colors):
 
 @ti.kernel
 def copy_image_f32_to_rgba8(src: ti.template(), dst: ti.template(),
-                            num_components: ti.template()):
-    for i, j in src:
+                            num_components: ti.template(),
+                            gray_scale: ti.template()):
+    for i, j in ti.ndrange(src.shape[0], src.shape[1]):
         px = ti.Vector([0, 0, 0, 0xff], dt=u32)
-        for k in ti.static(range(num_components)):
-            c = src[i, j][k]
+        if ti.static(gray_scale):
```

Comment:
Is there a better way not to use `if else` in `for` loops?

Reply:
Thanks for reviewing! I'm using static branching here, so I guess this `if else` statement will be eliminated compile-time leading to no runtime overhead? The document mentioned it [here](https://docs.taichi-lang.org/docs/meta#compile-time-branching).